ssl/t1_lib.c

```diff
@@ -2505,19 +2505,34 @@ static int has_usable_cert(SSL *s, const SIGALG_LOOKUP *sig, int idx)
             lu = tls1_lookup_sigalg(s->s3->tmp.peer_cert_sigalgs[i]);
             if (lu == NULL
                 || !X509_get_signature_info(s->cert->pkeys[idx].x509, &mdnid,
-                                            &pknid, NULL, NULL))
+                                            &pknid, NULL, NULL)
+                || mdnid != lu->hash
+                || pknid != lu->sig)
```

Comment:
I don't think that comment refers to any specific lines, does it? The loop as a whole fails to make that distinction because it would have to look at information which isn't currently available here.

Reply:
@dwmw2 is correct -- to be most accurate, we'd want to be looking at ```lu->sigalg``` instead of the separate hash and sig NIDs, but the necessary information isn't currently available here.
So while I don't expect any behavior change from moving this conditional, I do think the comment would need to be moved along with it (and reworded to clarify which check it applies to).

Reply:
Moved.